packages/nextclade/src/analyze/findNucChanges.cpp

```diff
@@ -7,10 +7,18 @@
 
 
 namespace Nextclade {
+  /**
+   * Finds nucleotide changes (nucleotide substitutions and deletions) as well
+   * as the beginning and end of the alignment range.
+   *
+   * @pre Precondition: sequences are expected to be stripped from insertions.
+   */
   NucleotideChangesReport findNucChanges(  //
     const NucleotideSequence& refStripped, //
     const NucleotideSequence& queryStripped//
   ) {
+    precondition_equal(refStripped.length(), queryStripped.length());
+
```

Comment:
Is this precondition correct? (not too strict, not too lax)

 - Not too strict: can there be a valid situation when this function encounters ref and query that are of different length?
 - Not too lax: can this function crash or produce invalid results even if the ref and query lengths are of the same length?